src/RandomInitialization.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Bdelespierre\Kmeans;
+
+use Bdelespierre\Kmeans\Cluster;
+use Bdelespierre\Kmeans\Interfaces\ClusterCollectionInterface;
+use Bdelespierre\Kmeans\Interfaces\InitializationSchemeInterface;
+use Bdelespierre\Kmeans\Interfaces\PointCollectionInterface;
+use Bdelespierre\Kmeans\Interfaces\PointInterface;
+use Bdelespierre\Kmeans\Point;
+
+class RandomInitialization implements InitializationSchemeInterface
+{
+    /**
+     * @throws \InvalidArgumentException when $nbClusters is lesser than 1
+     */
+    public function initializeClusters(PointCollectionInterface $points, int $nbClusters): ClusterCollectionInterface
+    {
+        // validate cluster count
+        if ($nbClusters < 1) {
+            throw new \InvalidArgumentException("Invalid cluster count: {$nbClusters}");
+        }
+
+        $clusters = new ClusterCollection($points->getSpace());
+
+        // get the space boundaries to avoid placing clusters centroid too far from points
+        list($min, $max) = $this->getBoundaries($points);
+
+        // initialize N clusters with a random point within space boundaries
+        for ($n = 0; $n < $nbClusters; $n++) {
+            // assign all points to the first cluster only
+            $clusters->add(new Cluster($this->getRandomPoint($min, $max), $n == 0 ? $points : null));
+        }
+
+        return $clusters;
+    }
+
+    /**
+     * @return array{PointInterface, PointInterface}
+     * @throws \InvalidArgumentException when $points is empty
+     */
+    protected function getBoundaries(PointCollectionInterface $points): array
```

Comment:
Isn't it okay to be private?

Reply:
It could be. But I prefer to leave those protected so they can be overridden if needed. If someone wants to keep the random initialization logic but has a different way to compute space boundaries (for non-euclidean geometry for example), they should be able to do so.